apps/mobile/src/features/ExecuteTx/components/ExecuteTransaction.tsx

```diff
@@ -0,0 +1,51 @@
+import { LoadingScreen } from '@/src/components/LoadingScreen'
+import React, { useEffect } from 'react'
+import { useLocalSearchParams } from 'expo-router'
+import { ExecuteError } from './ExecuteError'
+import { ExecuteSuccess } from './ExecuteSuccess'
+import { ExecutionStatus, useTransactionExecution } from '../hooks/useTransactionExecution'
+import { useAppSelector } from '@/src/store/hooks'
+import { selectActiveSigner } from '@/src/store/activeSignerSlice'
+import { useDefinedActiveSafe } from '@/src/store/hooks/activeSafe'
+import { useTransactionGuard } from '@/src/hooks/useTransactionGuard'
+
+export function ExecuteTransaction() {
+  const { txId } = useLocalSearchParams<{ txId: string }>()
+  const activeSafe = useDefinedActiveSafe()
+  const activeSigner = useAppSelector((state) => selectActiveSigner(state, activeSafe.address))
+  const { guard: canExecute } = useTransactionGuard('executing')
+  const { status, executeTx, retry } = useTransactionExecution({
+    txId: txId || '',
+    signerAddress: activeSigner?.value || '',
+  })
+
+  useEffect(() => {
+    if (canExecute && status === ExecutionStatus.IDLE && txId && activeSigner) {
+      executeTx()
+    }
+  }, [canExecute, status, executeTx, txId, activeSigner])
+
+  if (!txId) {
+    const handleRetry = () => {
+      console.error('Cannot retry: missing transaction ID')
+    }
+    return <ExecuteError description="Missing transaction ID" onRetryPress={handleRetry} />
+  }
+
+  if (!activeSigner) {
```

Comment:
how did the user end up with no active signer?

Reply:
As discussed we should clean up the activeSigner when the private key is removed. I will also update the type to reflect that activeSigner can be undefined and we can display a message to the user in that case.